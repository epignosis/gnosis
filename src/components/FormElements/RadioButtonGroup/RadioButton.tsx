import React, { FC, ReactNode, MouseEvent } from "react";
import classNames from "classnames";
import { SerializedStyles } from "@emotion/react";
import { InputSize } from "../Input/Input";
import { radioButton } from "./styles";

export type RadioButtonProps = {
  index: number;
  label: ReactNode;
  value: string;
};

export type UiRadioButtonProps = RadioButtonProps & {
  selectedValue: string;
  onClick: (value: string) => void;
  size: InputSize;
};

const RadioButton: FC<UiRadioButtonProps> = ({
  index,
  label,
  value,
  selectedValue,
  onClick,
  size,
}) => {
  const handleClick = (e: MouseEvent): void => {
    e.preventDefault();
    onClick(value);
  };
  const cssClasses = classNames({
    "is-selected": value === selectedValue,
  });

  return (
    <button
      id={`radio-${index}`}
      className={cssClasses}
      onClick={handleClick}
      css={(theme): SerializedStyles => radioButton(theme, { size })}
      aria-selected={value === selectedValue}
    >
      {label}
    </button>
  );
};

export default RadioButton;
