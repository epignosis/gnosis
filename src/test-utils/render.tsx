import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
