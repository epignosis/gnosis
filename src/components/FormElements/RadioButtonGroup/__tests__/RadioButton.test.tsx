import React from "react";
import userEvent from "@testing-library/user-event";
import faker from "faker";
import RadioButton from "../RadioButton";
import { screen, render } from "@test-utils/render";

describe("<RadioButton />", () => {
  it("renders correctly", () => {
    const labelTxt = faker.lorem.words();
    render(
      <RadioButton
        label={labelTxt}
        index={0}
        size="md"
        value={faker.random.alphaNumeric()}
        selectedValue=""
        onClick={jest.fn()}
      />,
    );

    const button = screen.getByText(labelTxt);

    expect(button).toHaveTextContent(labelTxt);
  });

  it("fires the onClick handler", () => {
    const mockOnClick = jest.fn();
    const labelTxt = faker.lorem.words();
    const value = faker.random.alphaNumeric();
    render(
      <RadioButton
        label={labelTxt}
        index={0}
        size="md"
        value={value}
        selectedValue=""
        onClick={mockOnClick}
      />,
    );

    const button = screen.getByText(labelTxt);

    userEvent.click(button);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
    expect(mockOnClick).toHaveBeenLastCalledWith(value);
  });

  it("matches snapshot with `size = md`", () => {
    const { container } = render(
      <RadioButton
        id="my-radio-id"
        className="my-radio-class"
        label="Test label"
        index={0}
        size="md"
        value="testValue"
        selectedValue=""
        onClick={jest.fn()}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with `size = lg`", () => {
    const { container } = render(
      <RadioButton
        label="Test label"
        index={0}
        size="lg"
        value="testValue"
        selectedValue=""
        onClick={jest.fn()}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
