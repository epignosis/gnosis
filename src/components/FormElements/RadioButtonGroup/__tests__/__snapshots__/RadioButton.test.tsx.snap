// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> matches snapshot with \`size = lg\` 1`] = `
<div>
  <button
    aria-selected="false"
    class=" css-1f7doby-radioButton-radioButton"
    id="radio-0"
  >
    Test label
  </button>
</div>
`;

exports[`<RadioButton /> matches snapshot with \`size = md\` 1`] = `
<div>
  <button
    aria-selected="false"
    class="my-radio-class css-19fta22-radioButton-radioButton"
    id="my-radio-id"
  >
    Test label
  </button>
</div>
`;
